### PR TITLE
BUGFIX: fix SearchedMultimodalDoc in docarray

### PR DESCRIPTION
**Pull request by BaoHuiling**
## Description

The summary of the proposed changes as long as the relevant motivation and context.

## Issues

Fix the conflict definition of ImageDoc, TextImageDoc and SearchedMultimodalDoc. 
Issue mentioned in comment: https://github.com/opea-project/GenAIComps/pull/496#issuecomment-2317686093
This change impact below PR: #496 #539 #575

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Comment by BaoHuiling** [comment]
@tileintel @s-gobriel
Please see the final version of SearchedMultimodalDoc.

**Comment by BaoHuiling** [comment]
@XuhuiRen  @lvliang-intel  Please help to review this patch, we have several PRs depending on this aligned change. Much appreciated!

**Review by XuhuiRen — APPROVED** [review]

File: comps/cores/proto/docarray.py [hunk 1]
```diff
@@ -110,14 +102,8 @@ class Config:
...
-class SearchedMultimodalDoc(BaseDoc):
-    retrieved_docs: List[TextImageDoc]
-    initial_query: str
-    top_n: int = 1
-    metadata: Optional[List[Dict]] = None
-
-    class Config:
-        json_encoders = {np.ndarray: lambda x: x.tolist()}
+class SearchedMultimodalDoc(SearchedDoc):
+    metadata: List[Dict[str, Any]]
```
Comment (tileintel) [inline comps/cores/proto/docarray.py:106]: @BaoHuiling Should we have `metadata: List[Dict[str, Any]] = None`?

Comment (BaoHuiling) [reply comps/cores/proto/docarray.py:106]: That will make this field optional, then why user not use SearchedDoc directly?

Comment (tileintel) [reply comps/cores/proto/docarray.py:106]: Agree on optional. However, using SearchedDoc directly without metadata variable cannot enforce future development to make use of the variable `metadata`. One can simply extends SearchedDoc with additional image, audio, etc... for the purpose of multimodality, making this difficult to inherit and reuse in many other downstream microservice.   
It is ok to leave it without `None`. I think the results will be the same.

Comment (BaoHuiling) [reply comps/cores/proto/docarray.py:106]: Agree. If we need it in the future, issue another PR is not complicated.

**Comment by BaoHuiling** [comment]
Hello Liang, any comment on the PR? @lvliang-intel

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**